File: config/config_loader.py
import json
import os

def load_config(env: str = None):
    env = env or os.getenv("TEST_ENV", "dev")
    config_path = os.path.join(os.path.dirname(__file__), f"{env}.json")

    with open(config_path, "r") as f:
        return json.load(f)

File: playwright/utils/configLoader.ts
import fs from 'fs';
import path from 'path';

export function loadConfig(env: string) {
  const filePath = path.join(__dirname, `../configs/config.${env}.json`);

  if (!fs.existsSync(filePath)) {
    throw new Error(`Config file not found for environment: ${env}`);
  }

  const raw = fs.readFileSync(filePath, 'utf-8');
  return JSON.parse(raw);
}